lib/puppet/util/windows/file.rb

```diff
@@ -257,6 +257,45 @@ def readlink(link_name)
   end
   module_function :readlink
 
+  ERROR_FILE_NOT_FOUND = 2
+  ERROR_PATH_NOT_FOUND = 3
+
+  def get_long_pathname(path)
+    converted = ''
+    FFI::Pointer.from_string_to_wide_string(path) do |path_ptr|
+      # includes terminating NULL
+      buffer_size = GetLongPathNameW(path_ptr, FFI::Pointer::NULL, 0)
+      FFI::MemoryPointer.new(:wchar, buffer_size) do |converted_ptr|
+        if GetLongPathNameW(path_ptr, converted_ptr, buffer_size) == FFI::WIN32_FALSE
+          raise Puppet::Util::Windows::Error.new("Failed to call GetLongPathName")
+        end
+
+        converted = converted_ptr.read_wide_string(buffer_size - 1)
+      end
+    end
+
+    converted
+  end
+  module_function :get_long_pathname
+
+  def get_short_pathname(path)
+    converted = ''
+    FFI::Pointer.from_string_to_wide_string(path) do |path_ptr|
+      # includes terminating NULL
+      buffer_size = GetShortPathNameW(path_ptr, FFI::Pointer::NULL, 0)
+      FFI::MemoryPointer.new(:wchar, buffer_size) do |converted_ptr|
+        if GetShortPathNameW(path_ptr, converted_ptr, buffer_size) == FFI::WIN32_FALSE
+          raise Puppet::Util::Windows::Error.new("Failed to call GetShortPathName")
+        end
+
+        converted = converted_ptr.read_wide_string(buffer_size - 1)
+      end
+    end
+
+    converted
+  end
+  module_function :get_short_pathname
```

Comment:
Yeah, I'm glad that is does also have its own tests, otherwise I would have voted for the removal until needed.